sw/device/tests/dif/dif_rstmgr_unittest.cc

```diff
@@ -0,0 +1,306 @@
+  for (uint32_t bit_index = 0; bit_index < RSTMGR_PARAM_NUMSWRESETS;
+       ++bit_index) {
+    uint32_t bitfield = bitfield_bit32_write(
+        std::numeric_limits<uint32_t>::max(), bit_index, false);
+    EXPECT_WRITE32(RSTMGR_SW_RST_REGEN_REG_OFFSET, bitfield);
+
+    EXPECT_EQ(dif_rstmgr_reset_lock(&rstmgr_, bit_index), kDifRstmgrOk);
+  }
+}
+
+class ResetIsLockedTest : public RstmgrTest {};
+
+TEST_F(ResetIsLockedTest, NullArgs) {
+  bool is_locked;
+  EXPECT_EQ(dif_rstmgr_reset_is_locked(nullptr, 0, nullptr), kDifRstmgrBadArg);
+  EXPECT_EQ(dif_rstmgr_reset_is_locked(nullptr, 0, &is_locked),
+            kDifRstmgrBadArg);
+  EXPECT_EQ(dif_rstmgr_reset_is_locked(&rstmgr_, 0, nullptr), kDifRstmgrBadArg);
+}
+
+TEST_F(ResetIsLockedTest, BadPeripheral) {
+  bool is_locked;
+  EXPECT_EQ(dif_rstmgr_reset_is_locked(&rstmgr_, RSTMGR_PARAM_NUMSWRESETS,
+                                       &is_locked),
+            kDifRstmgrBadArg);
+}
+
+TEST_F(ResetIsLockedTest, Success) {
+  for (uint32_t bit_index = 0; bit_index < RSTMGR_PARAM_NUMSWRESETS;
+       ++bit_index) {
+    uint32_t bit_locked = bitfield_bit32_write(
+        std::numeric_limits<uint32_t>::max(), bit_index, false);
+    EXPECT_READ32(RSTMGR_SW_RST_REGEN_REG_OFFSET, bit_locked);
+
+    bool is_locked = false;
+    EXPECT_EQ(dif_rstmgr_reset_is_locked(&rstmgr_, bit_index, &is_locked),
+              kDifRstmgrOk);
+    EXPECT_TRUE(is_locked);
+
+    is_locked = true;
+    EXPECT_READ32(RSTMGR_SW_RST_REGEN_REG_OFFSET, {{bit_index, true}});
+    EXPECT_EQ(dif_rstmgr_reset_is_locked(&rstmgr_, bit_index, &is_locked),
+              kDifRstmgrOk);
+    EXPECT_FALSE(is_locked);
+  }
+}
+
+class ResetCausesGetTest : public RstmgrTest {
+ protected:
+  ResetCausesGetTest() {
+    // Make sure that the test is up-to-date with the implementation.
+    EXPECT_EQ(0x1 << reset_info_reasons_.back(), kDifRstmgrResetInfoLast);
+  }
+
+  const std::vector<uint32_t> reset_info_reasons_{
+      RSTMGR_RESET_INFO_POR,
+      RSTMGR_RESET_INFO_LOW_POWER_EXIT,
+      RSTMGR_RESET_INFO_NDM_RESET,
+      RSTMGR_RESET_INFO_HW_REQ,
+  };
+};
+
+TEST_F(ResetCausesGetTest, NullArgs) {
+  dif_rstmgr_reset_info_bitfield_t info;
+  EXPECT_EQ(dif_rstmgr_reset_info_get(nullptr, nullptr), kDifRstmgrBadArg);
+  EXPECT_EQ(dif_rstmgr_reset_info_get(nullptr, &info), kDifRstmgrBadArg);
+  EXPECT_EQ(dif_rstmgr_reset_info_get(&rstmgr_, nullptr), kDifRstmgrBadArg);
+}
+
+TEST_F(ResetCausesGetTest, Success) {
+  // Single reason expectations.
+  for (auto reason : reset_info_reasons_) {
+    uint32_t bitfield = bitfield_bit32_write(0, reason, true);
+    EXPECT_READ32(RSTMGR_RESET_INFO_REG_OFFSET, bitfield);
+
+    dif_rstmgr_reset_info_bitfield_t info;
+    EXPECT_EQ(dif_rstmgr_reset_info_get(&rstmgr_, &info), kDifRstmgrOk);
+    EXPECT_EQ(info & bitfield, info);
+  }
+
+  // The first and the last reset causes.
+  uint32_t reasons = 0x1 << reset_info_reasons_.front();
+  reasons |= (0x1 << reset_info_reasons_.back());
```

Comment:
I think you can just use EXPECT_READ's `{}` syntax here?

Reply:
Good point, I will try that.

Reply:
Done.